### PR TITLE
Fix test classpath

### PR DESCRIPTION
**Pull request by madisp**
Fixes test classpath by placing idea dependencies on compile (and
runtime) dependencies instead of compileOnly.

### CODE REVIEW

File: src/main/groovy/org/jetbrains/intellij/IntelliJPlugin.groovy [hunk 1]
```diff
@@ -71,7 +71,7 @@ class IntelliJPlugin implements Plugin<Project> {
 
         if (JavaPlugin.hasProperty('COMPILE_ONLY_CONFIGURATION_NAME')) {
             project.configurations.getByName(JavaPlugin.COMPILE_ONLY_CONFIGURATION_NAME).extendsFrom idea, ideaPlugins
-            project.configurations.getByName(JavaPlugin.TEST_COMPILE_ONLY_CONFIGURATION_NAME).extendsFrom idea, ideaPlugins
+            project.configurations.getByName(JavaPlugin.TEST_COMPILE_CONFIGURATION_NAME).extendsFrom idea, ideaPlugins
```
Comment (zolotov) [inline src/main/groovy/org/jetbrains/intellij/IntelliJPlugin.groovy]: what's wrong with `TEST_COMPILE_ONLY_CONFIGURATION_NAME` here?

At least `ideaPlugins` should not be in test runtime

Comment (madisp) [inline src/main/groovy/org/jetbrains/intellij/IntelliJPlugin.groovy]: hmm, I had problems where it couldn't find JUnit on the classpath but I can't reproduce anymore. Maybe I had some other configuration issue at the same time? Feel free to reject the PR.

Comment (madisp) [inline src/main/groovy/org/jetbrains/intellij/IntelliJPlugin.groovy]: here are the classpath differences for a project that doesn't depend on any plugins: https://gist.github.com/madisp/ea0b0793363795a73f845b27ebe8c8d2

Comment (zolotov) [inline src/main/groovy/org/jetbrains/intellij/IntelliJPlugin.groovy]: Oh, I see. Then let's keep `idea` in classpath and exclude `ideaPlugins` from it.

Test Fixture loads all plugins from `idea.plugins.path` property and each plugin have its own ClassLoader to load classes from a corresponding directory, so having them in classpath look redundant at least.

**Comment by zolotov** [comment]
closing PR, implemented in master

**Pull request closed without merging**